crates/biome_grit_parser/src/parser/definitions.rs

```diff
@@ -0,0 +1,117 @@
+use super::patterns::parse_pattern_list;
+use super::predicates::parse_predicate_list;
+use super::{parse_language_declaration, parse_name, parse_variable_list, GritParser};
+use biome_grit_syntax::{GritSyntaxKind::*, T};
+use biome_parser::prelude::ParsedSyntax::*;
+use biome_parser::{parsed_syntax::ParsedSyntax, Parser};
+
+pub(crate) fn parse_definition_list(p: &mut GritParser) -> ParsedSyntax {
```

Comment:
For lists, the parsing infrastructure offer a `ParseNodeList` and `ParseSeparatedList`. The trait exposes a `parse_list` method, which we **must** use. Internally, this method does some nice checks: it checks if the parser progresses while parsing each item of the list.